[AMORO-3671][Improvement]: Optimizing Allocation and Schedule Priority of Optimizer resource for Tables

cxxiii opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://amoro.apache.org/how-to-contribute/
  2. If the PR is related to an issue in https://github.com/apache/amoro/issues, add '[AMORO-XXXX]' in your PR title, e.g., '[AMORO-XXXX] Your PR title ...'.
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][AMORO-XXXX] Your PR title ...'.
-->

## Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you add a feature, you can talk about its use case.
  2. If you fix a bug, you can clarify why it is a bug.
  3. Use Fix/Resolve/Close #{ISSUE_NUMBER} to link this PR to its related issue
-->

Close #3671 .

## Brief change log
<!--
Clearly describe the changes made in modules, classes, methods, etc.
-->

- Refine the definition of configuration parameter Quota and the calculation of metric Occupation—as well as improving the current self-optimizing scheduling policy based on Quota.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

zhoujinsong reviewed (commented):
Hi, I left some comments.

Besides, can you add some unit test cases for this new feature?
And some docs should be updated too.

cxxiii commented:
> Hi, I left some comments.
> 
> Besides, can you add some unit test cases for this new feature? And some docs should be updated too.

Thank you for your suggestions. I added some unit test cases and updated the related doc, please review again.

cxxiii commented:
> I left some comments, PTAL.

Thanks for your detailed review. I have made the changes accordingly.

zhoujinsong commented on amoro-ams/src/main/java/org/apache/amoro/server/optimizing/OptimizingProcess.java:
```diff
@@ -24,6 +24,8 @@
 
 public interface OptimizingProcess {
 
+  int getQuotaLimit();
```
Maybe we should not add this method to the interface, it is only used inner the implementation.

cxxiii replied:
ok, thanks for your suggestions.
